lib/actions/environment-variables.coffee

```diff
@@ -31,19 +34,31 @@ exports.list =
 	]
 	permission: 'user'
 	action: (params, options, done) ->
-		resin.models.environmentVariables.getAllByApplication options.application, (error, environmentVariables) ->
-			return done(error) if error?
+		async.waterfall([
```

Comment:
This would be nicer if you made use of the models promise interface, ie
```coffee
Promise.try ->
	if options.application?
		resin.models.environmentVariables.getAllByApplication(options.application, callback)
	else if options.device?
		resin.models.environmentVariables.device.getAll(options.device, callback)
	else
		throw new Error('You must specify an application or device')
.then (environmentVariables) ->
	if not options.verbose
		isSystemVariable = resin.models.environmentVariables.isSystemVariable
		environmentVariables = _.reject(environmentVariables, isSystemVariable)

	console.log visuals.widgets.table.horizontal environmentVariables, [
		'id'
		'name'
		'value'
	]
.nodeify(done)
```